src/components/actions/dev_tools/databaseMethods.ts

```diff
@@ -0,0 +1,248 @@
+import { Decision } from '../../../generated/graphql'
+import DBConnect from '../../databaseConnect'
+
+const databaseMethods = {
+  // Gets the application/template info of the "config" application associated with the template
+  // specified by "templateCode". If multiple versions of the template exist, it
+  // will use the one marked "AVAILABLE" or, if none are available, the highest
+  // version number.
+  getConfigApplicationInfo: async (templateCode: string) => {
+    const text = `
+    WITH available_count AS (
+      SELECT COUNT(*) FROM template
+      WHERE code = $1
+      AND status = 'AVAILABLE'
+    )
+    SELECT app.id as "configId", app.serial as "configSerial",
+    app.template_id as "templateId", ts."sectionCodes"
+    FROM application app
+    JOIN (
+      SELECT template_id, array_agg(code) as "sectionCodes" from template_section
+      GROUP BY template_id
+    ) ts
+    ON ts.template_id = app.template_id
+    WHERE is_config=true
+    AND app.template_id = (
+      SELECT id FROM template
+      WHERE code = $1
+      AND CASE
+        WHEN (SELECT count FROM available_count) = 0 THEN version = (
+        SELECT MAX(version) from template
+        WHERE code = $1
+      ) ELSE status = 'AVAILABLE'
+      END
+    )`
+    try {
+      const result = await DBConnect.query({ text, values: [templateCode] })
+      return result.rows[0] ?? {}
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  getSerialFromAppId: async (applicationId: number) => {
+    const text = `
+      SELECT serial FROM application
+      WHERE id = $1;
+    `
+    try {
+      const result = await DBConnect.query({ text, values: [applicationId] })
+      console.log('result', result.rows)
+      return result.rows[0].serial
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  getAppIdFromSerial: async (serial: string) => {
+    const text = `
+      SELECT id FROM application
+      WHERE serial = $1;
+    `
+    try {
+      const result = await DBConnect.query({ text, values: [serial] })
+      console.log('result', result.rows)
+      return result.rows[0].id
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  getSingleReviewAssignment: async (assignmentId: number) => {
+    const text = `
+      SELECT id, allowed_sections AS "allowedSections"
+      FROM review_assignment WHERE id = $1
+    `
+    try {
+      const result = await DBConnect.query({ text, values: [assignmentId] })
+      return result.rows[0]
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  // Gets all the review assignments for the specified application at the
+  // highest possible stage and review_level
```

Comment:
Just wondering if it would be possible to also do partial reviews using this tool. For example, could this check also for Review to be created in the current stage and level if there are un-assigned questions?

Reply:
> For example, could this check also for Review to be created in the current stage and level if there are un-assigned questions?

This tool doesn't check anything -- it just simulates events. It's up to you to configure the state of things if you want to test specific scenarios. Which you can do by either creating the applications for real (the slow but probably more reliable way), or simulating it by overriding applicationData in your endpoint request here (fast, but possibly not 100% realistic)